app/src/main/java/de/danoeh/antennapod/fragment/swipeactions/MarkFavourite.java

```diff
@@ -0,0 +1,43 @@
+package de.danoeh.antennapod.fragment.swipeactions;
+
+import android.content.Context;
+
+import androidx.fragment.app.Fragment;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.adapter.actionbutton.DownloadActionButton;
+import de.danoeh.antennapod.core.storage.DBWriter;
+import de.danoeh.antennapod.model.feed.FeedItem;
+import de.danoeh.antennapod.model.feed.FeedItemFilter;
+
+public class MarkFavourite extends SwipeAction {
+
+    @Override
+    public int actionIcon() {
+        return R.drawable.ic_star;
+    }
+
+    @Override
+    public int actionColor() {
+        return R.color.swipe_yellow_200;
+    }
+
+    @Override
+    public String title(Context context) {
+        return context.getString(R.string.add_to_favorite_label);
+    }
+
+    @Override
+    public void action(FeedItem item, Fragment fragment, FeedItemFilter filter) {
+        DBWriter.toggleFavoriteItem(item);
+    }
+
+    @Override
+    List<String> affectedFilters() {
+        return Arrays.asList("is_favorite", "not_favorite");
```

Comment:
This is risky. When changing the string constants inside the filters, these here will likely be forgotten. Instead, please just pass the filter as a parameter and use the filter's attributes to decide wether this is affected.

Reply:
So should I use R.string? Why aren't the properties constants in e.g. the FeedItemFilter class? There they are used as raw strings as well which is just a risky!

Reply:
> So should I use R.string?

I would just use the filter object directly instead of dealing with its internal strings (independently of where they are defined).

Reply:
Which object do you mean? I use the FeedItemFilter from the fragments, but I do have to check for each SwipeAction which filters are affected, how should I get these strings?

Reply:
like this?

Reply:
Yeah, thanks!